diff -urN sap-0.2.old/src/configuration.h sap-0.2/src/configuration.h
--- sap-0.2.old/src/configuration.h	Fri Jul 27 04:34:44 2001
+++ sap-0.2/src/configuration.h	Thu Jan 10 22:16:22 2002
@@ -6,10 +6,17 @@
 #define MSB_FIRST_PROCESSOR
 */
 
+#ifndef DICTIONARIES_DIRECTORY
 #define DICTIONARIES_DIRECTORY "/usr/lib/sap/"
+#endif
+
 #define DIRECTORY_SEPARATOR "/"
 #define LOCAL_SAPRC_FILENAME ".saprc"
+
+#ifndef GLOBAL_SAPRC_FULLPATH
 #define GLOBAL_SAPRC_FULLPATH "/etc/saprc"
+#endif
+
 #define FILENAME_DATABASE_1 "dvp_1.dic"
 #define FILENAME_DATABASE_2 "dvp_2.dic"
 
